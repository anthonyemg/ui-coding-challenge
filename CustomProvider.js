import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducer from './src/public/reducers';

const store = createStore(reducer);

const CustomProvider = ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};

export default CustomProvider;
