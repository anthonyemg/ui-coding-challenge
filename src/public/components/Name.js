import React from 'react';
import { connect } from 'react-redux';
import { updateName, showNameModal, hideNameModal } from '../actions';
import Modal from './Modal';
import NameModal from './NameModal';

class Name extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      updatedName: ''
    };
    this.handleUpdateName = this.handleUpdateName.bind(this);
    this.handleSubmitUpdatedName = this.handleSubmitUpdatedName.bind(this);
  }
  handleUpdateName(e) {
    this.setState({
      updatedName: e.target.value
    });
  }
  handleSubmitUpdatedName() {
    this.props.updateName(this.state.updatedName);
  }
  render() {
    return (
      <div className="component-container">
        <div className="component-wrapper">
          <div className="component-details">
            <span className="component-title">Name</span>
            <span className="component-text">{this.props.name}</span>
          </div>
          <div>
            <button
              className="component-editButton"
              onClick={() => this.props.showNameModal()}
            >
              Edit Name
            </button>
          </div>
        </div>
        {this.props.displayNameModal && (
          <Modal>
            <NameModal
              name={this.props.name}
              updateName={this.props.updateName}
              handleUpdateName={this.handleUpdateName}
              handleSubmitUpdatedName={this.handleSubmitUpdatedName}
              hideNameModal={this.props.hideNameModal}
            />
          </Modal>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  name: state.name,
  displayNameModal: state.displayNameModal
});
const mapDispatchToProps = dispatch => ({
  updateName: name => dispatch(updateName(name)),
  showNameModal: () => dispatch(showNameModal()),
  hideNameModal: () => dispatch(hideNameModal())
});

export default connect(mapStateToProps, mapDispatchToProps)(Name);
