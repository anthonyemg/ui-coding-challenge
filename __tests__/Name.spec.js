import React from 'react';
import renderer from 'react-test-renderer';
import CustomProvider from '../CustomProvider';
import Name from '../src/public/components/Name';

test('the Name component renders correctly', () => {
  const component = renderer.create(
    <CustomProvider>
      <Name />
    </CustomProvider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
