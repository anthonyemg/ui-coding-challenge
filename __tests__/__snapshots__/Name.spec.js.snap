// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Name component renders correctly 1`] = `
<div
  className="component-container"
>
  <div
    className="component-wrapper"
  >
    <div
      className="component-details"
    >
      <span
        className="component-title"
      >
        Name
      </span>
      <span
        className="component-text"
      >
        Anthony G
      </span>
    </div>
    <div>
      <button
        className="component-editButton"
        onClick={[Function]}
      >
        Edit Name
      </button>
    </div>
  </div>
</div>
`;
